src/cargo/ops/cargo_fetch.rs

```diff
@@ -1,25 +1,23 @@
 use std::path::Path;
 
 use core::registry::PackageRegistry;
-use core::{Source, PackageId};
+use core::{Package, PackageId, Resolve};
 use ops;
-use sources::PathSource;
 use util::{CargoResult, Config, human, ChainError};
 
 /// Executes `cargo fetch`.
 pub fn fetch(manifest_path: &Path, config: &Config) -> CargoResult<()> {
-    let mut source = try!(PathSource::for_path(manifest_path.parent().unwrap(),
-                                               config));
-    try!(source.update());
-    let package = try!(source.root_package());
-
+    let package = try!(Package::for_path(manifest_path, config));
     let mut registry = PackageRegistry::new(config);
-    registry.preload(package.package_id().source_id(), Box::new(source));
     let resolve = try!(ops::resolve_pkg(&mut registry, &package));
+    let _ = get_resolved_packages(&resolve, &mut registry);
+    Ok(())
+}
 
+pub fn get_resolved_packages(resolve: &Resolve, registry: &mut PackageRegistry)
+                             -> CargoResult<Vec<Package>> {
     let ids: Vec<PackageId> = resolve.iter().cloned().collect();
-    try!(registry.get(&ids).chain_error(|| {
-        human("unable to get packages from source")
-    }));
-    Ok(())
+    registry.get(&ids).chain_error(|| {
+        human("Unable to get packages from source")
```

Comment:
Could this stay as lowercase? (conventionally our errors start with lowercase letters)